custom-checks/pmd/src/main/java/org/openhab/tools/analysis/pmd/UseSLF4JLoggerRule.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.pmd;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceType;
+import net.sourceforge.pmd.lang.java.ast.ASTReferenceType;
+import net.sourceforge.pmd.lang.java.ast.ASTType;
+import net.sourceforge.pmd.lang.java.ast.ASTVariableDeclarator;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
+
+/**
+ * Checks if a logger other than the one provided by slf4j is used
+ *
+ * @author Lyubomir Papazov - Initial contribution
+ *
+ */
+public class UseSLF4JLoggerRule extends AbstractJavaRule {
+
+    //There is a very similar check created by PMD that checks if only one logger is used:
+    //https://github.com/pmd/pmd/blob/pmd/5.8.x/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/logging/MoreThanOneLoggerRule.java
```

Comment:
Can you add a description what is different from the one you mention and why you had to create this one.

Is the license btw ok to reuse the code like this.

Reply:
I added a few more comment lines to explain why this check is needed. 

I used the pmd [check](https://github.com/pmd/pmd/blob/pmd/5.8.x/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/logging/MoreThanOneLoggerRule.java) as a reference because it was very close to my use case but the class doesn't seem to be designed to be extendable. After I've done the refactoring that you suggested this class doesn't seem too similar to the original and it also serves a different purpose.